src/NuGet.Core/NuGet.Protocol/HttpSource/HttpRetryHandler.cs

```diff
@@ -270,5 +301,22 @@ internal HttpRetryHandler(IEnvironmentVariableReader environmentVariableReader)
 
             return response;
         }
+
+        private static TimeSpan? GetRetryAfter(RetryConditionHeaderValue? retryAfter)
+        {
+            if (retryAfter?.Delta != null)
+            {
+                return retryAfter.Delta;
+            }
+
+            if (retryAfter?.Date != null)
+            {
+                DateTimeOffset retryAfterDate = retryAfter.Date.Value.ToUniversalTime();
+                var now = DateTimeOffset.UtcNow;
+                return retryAfterDate - now;
```

Comment:
I learnt from this [doc](https://developer.mozilla.org/docs/Web/HTTP/Headers/Retry-After) that `RetryAfter` may not null even for `503` response. Should we have a maximum threshold value for `RetryAfter` when the date or delay is beyond acceptable limits?

Reply:
Let's ignore malicious servers, because there are other ways that they could be more disruptive to clients. Let's also ignore malicious customers because there are much easier ways to attempt denial of service attacks.

If a server says it's busy and please don't retry for x duration, wouldn't NuGet be a bad netizen for retrying more quickly?

There is a scenario where the server is poorly configured and sending Retry-After durations that are longer than they need to be. In this case, being able to limit the max wait duration could help customers restore their projects more quickly.

While we certainly have a responsibility to customers using our software to help them unblock themselves in bad scenarios, I think we also have a responsibility to server admins to minimize risk that our software hammers their services. So, there's a balance, but I don't know what the best solution is.

Maybe allow customers to set a max retry duration, but don't let customers set it to less than 30 seconds? 10 seconds?  I'm not sure what the lower bound we should allow.  Maybe the minimum wait time should be equal to the exponential delay (but since they can set it to zero milliseconds, there's already an example of how NuGet isn't even attempting to protect server admins), so NuGet won't retry more quickly than if Retry-After wasn't implemented at all.

I'm sure I'm overthinking this, but I just don't feel like there's an obvious answer. No matter what we choose, there are some people (customers who can't mitigate bad servers, or server admins whose servers are getting hammered by clients who don't implement the HTTP spec) that will be negatively impacted.

Reply:
I don't really know what we choose, but I feel like we *have* to choose something. 

Is 2 mins too much?

Reply:
I pushed a change. Hopefully our customers aren't jerks, but there are easier ways to send too much traffic to a server, so I guess it's low risk.